shared/transforms/stig_refs.sh

```diff
@@ -0,0 +1,77 @@
+			VRELEASE=$(awk "/<Group id=\"${VKEY}\"/,/<\/Group/" $XCCDF_FILE | grep -i 'Rule id=' | sed -e 's/.*Rule id="SV-[0-9]*//' -e 's/r\([0-9]\).*/\1/')
+			IACONTROLS=$(awk "/<Group id=\"${VKEY}\"/,/<\/Group/" $XCCDF_FILE | grep "&lt;IAControls&gt;" | sed -e 's/.*&lt;IAControls&gt;//' -e 's/&.*//' -e 's/,[ ]*/,/g')
+			TITLE=$(awk "/<Group id=\"${VKEY}\"/,/<\/Group/" $XCCDF_FILE | awk "/<Rule /,/<\/Rule>/" | grep '<title>' | sed -e "s/.*<title>//" -e "s/<\/title>.*//")
+			RULE_ID=$(grep "<overlay.*ownerid=\"${STIG_ID}\"" input/auxiliary/stig_overlay.xml | sed -e 's/.*ruleid="//' -e 's/".*//')
+			echo "${VKEY}|${STIG_ID}|${CCI}|${CCE}|${SEVERITY}|${SVKEY}|${VRELEASE}|${IACONTROLS}|${TITLE}|${RULE_ID}"
+			sed -i -e "/<overlay.*ownerid=\"${STIG_ID}\"/,/<\/overlay>/s/disa=\"[0-9]*\"/disa=\"$(echo ${CCI} | cut -d"," -f1)\"/" -e "/<overlay.*ownerid=\"${STIG_ID}\"/,/<\/overlay>/s/severity=\"[a-z]*\"/severity=\"${SEVERITY}\"/" -e "/<overlay.*ownerid=\"${STIG_ID}\"/,/<\/overlay>/s/VKey=\"[0-9]*\"/VKey=\"$(echo ${VKEY}|sed 's/[vV]-//')\"/" -e "/<overlay.*ownerid=\"${STIG_ID}\"/,/<\/overlay>/s/SVKey=\"[0-9]*\"/SVKey=\"${SVKEY}\"/" -e "/<overlay.*ownerid=\"${STIG_ID}\"/,/<\/overlay>/s/VRelease=\"[0-9]*\"/VRelease=\"${VRELEASE}\"/" -e "\|<overlay.*ownerid=\"${STIG_ID}\"|,\|</overlay>|s|<title>.*</title>|<title>${TITLE}</title>|" input/auxiliary/stig_overlay.xml
+			if [ "$(grep -R "<Rule id=\"${RULE_ID}\"" input/services input/system 2>/dev/null| grep -c ^)" != 0 ]; then
+				grep -R "<Rule id=\"${RULE_ID}\"" input/services input/system 2>/dev/null | cut -d: -f1 | uniq | while read FILE; do
+					# SEVERITY
+					if [ ! -z "${SEVERITY}" ]; then
+						sed -i -e "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/severity=\"[a-z]*\"/severity=\"${SEVERITY}\"/" ${FILE}
+					fi
+					# CCE
+					if [ ! -z "${CCE}" ]; then
+						if [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep -c "<ident ")" = "0" ]; then
+							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<\/Rule>\)/<ident cce=\"${CCE}\" \/>\n\1/" ${FILE}
+						elif [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep "<ident " | grep -c "cce=")" = "0" ]; then
+							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)\/>/\1 cce=\"${CCE}\" \/>/" ${FILE}
+						else
+							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)cce=\"[a-zA-Z0-9\-\_.]*\"/\1cce=\"${CCE}\"/" ${FILE}
+						fi
+					fi
+					# STIG ID
+					if [ ! -z "${STIG_ID}" ]; then
+						if [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep -c "<ident ")" = "0" ]; then
+							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<\/Rule>\)/<ident stig=\"${STIG_ID}\" \/>\n\1/" ${FILE}
+						elif [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep "<ident " | grep -c "stig=")" = "0" ]; then
+							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)\/>/\1 stig=\"${STIG_ID}\" \/>/" ${FILE}
```

Comment:
So I checked in a fix. It will no longer create the double spacing when
defining new values. However, running it won't fix the double spacing for
the ones already done. I do have a quick fix for that though. Since I need
to commit some additional changes for the RHEL6 STIG refs, I will also
commit the change to remove the double spacing added.

On Wed, Feb 25, 2015 at 12:17 PM, Trey Henefield <thenefield@gmail.com>
wrote:

> Will fix. Standby.
>
> On Wed, Feb 25, 2015 at 12:16 PM, Shawn Wells <notifications@github.com>
> wrote:
>
>> In shared/transforms/stig_refs.sh
>> <https://github.com/OpenSCAP/scap-security-guide/pull/452#discussion_r25363784>
>> :
>>
>> > +					# CCE
>> > +					if [ ! -z "${CCE}" ]; then
>> > +						if [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep -c "<ident ")" = "0" ]; then
>> > +							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<\/Rule>\)/<ident cce=\"${CCE}\" \/>\n\1/" ${FILE}
>> > +						elif [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep "<ident " | grep -c "cce=")" = "0" ]; then
>> > +							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)\/>/\1 cce=\"${CCE}\" \/>/" ${FILE}
>> > +						else
>> > +							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)cce=\"[a-zA-Z0-9\-\_.]*\"/\1cce=\"${CCE}\"/" ${FILE}
>> > +						fi
>> > +					fi
>> > +					# STIG ID
>> > +					if [ ! -z "${STIG_ID}" ]; then
>> > +						if [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep -c "<ident ")" = "0" ]; then
>> > +							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<\/Rule>\)/<ident stig=\"${STIG_ID}\" \/>\n\1/" ${FILE}
>> > +						elif [ "$(awk "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/" ${FILE} | grep "<ident " | grep -c "stig=")" = "0" ]; then
>> > +							sed -i "/<Rule id=\"${RULE_ID}\"/,/<\/Rule>/s/\(<ident .*\)\/>/\1 stig=\"${STIG_ID}\" \/>/" ${FILE}
>>
>> this actually inserts two spaces before the stig=$val, e.g.:
>>
>>  <ident cce="26910-0"  stig="RHEL-06-000282" />
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/OpenSCAP/scap-security-guide/pull/452/files#r25363784>
>> .
>>
>
>